app/controllers/suggestions_controller.rb

```diff
@@ -0,0 +1,91 @@
+class SuggestionsController < ApplicationController
+  skip_before_action :require_login, only: [:index, :show]
+
+  def index
+    choose_correct_suggestions
+  end
+
+  def show
+    choose_correct_suggestions
+    @search = params[:search]
+    @limit = params[:limit]
+
+    if @suggestions.nil?
+      redirect_to root_path, notice: "No matches for #{params["search"]}!"
+    else
+      render :show
+    end
+  end
+
+  def favorites
+    user_id = current_user.uid
+    fav_suggestions_ids = TunesTakeoutWrapper.favorites(user_id) #=> returns array of ids
+    fav_suggestions = TunesTakeoutWrapper.find_many(fav_suggestions_ids).suggestions
+    @suggestions = transform_suggestions(fav_suggestions)
+
+    render :show
+  end
+
+  def favorite
+    suggestion_id = params[:suggestion_id]["suggestion_id"]
+    user_id = current_user.uid
+    TunesTakeoutWrapper.favorite(user_id, suggestion_id)
+    redirect_to favorites_path
+  end
+
+  def unfavorite
+    suggestion_id = params[:suggestion_id]["suggestion_id"]
+    user_id = current_user.uid
+    TunesTakeoutWrapper.unfavorite(user_id, suggestion_id)
+    redirect_to favorites_path
+  end
+
+  private
+  def search_tunes_params
+    params.permit[:search]
+  end
+
+  def transform_suggestions(raw_suggestions)
```

Comment:
Oh awesome, I've been struggling with figuring out where things like that belong. So mostly controller methods should just be methods that map to routes then?